var/spack/repos/builtin/packages/geant4/package.py

```diff
@@ -15,84 +13,110 @@ class Geant4(CMakePackage):
+        depends_on('xerces-c cxxstd=' + std, when='cxxstd=' + std)
+
+        # Vecgeom specific versions for each Geant4 version
+        depends_on('vecgeom@1.1.5 cxxstd=' + std,
+                   when='@10.6.0:10.6.99 +vecgeom cxxstd=' + std)
+        depends_on('vecgeom@1.1.0 cxxstd=' + std,
+                   when='@10.5.0:10.5.99 +vecgeom cxxstd=' + std)
+        depends_on('vecgeom@0.5.2 cxxstd=' + std,
+                   when='@10.4.0:10.4.99 +vecgeom cxxstd=' + std)
+        depends_on('vecgeom@0.3rc cxxstd=' + std,
+                   when='@10.3.0:10.3.99 +vecgeom cxxstd=' + std)
+
+    # Visualization driver ependencies
     depends_on("gl", when='+opengl')
     depends_on("glx", when='+opengl+x11')
     depends_on("libx11", when='+x11')
     depends_on("libxmu", when='+x11')
     depends_on("motif", when='+motif')
-    depends_on("qt@4.8:", when="+qt")
-
-    # if G4 data not installed with geant4
-    # depend on G4 data packages
-    # this allows external data installations
-    # to avoid duplication
-
-    depends_on('geant4-data@10.03.p03', when='@10.03.p03 ~data')
-    depends_on('geant4-data@10.04', when='@10.04 ~data')
-    depends_on('geant4-data@10.05.p01', when='@10.05.p01 ~data')
+    depends_on("qt@5:", when="+qt")
 
     # As released, 10.03.03 has issues with respect to using external
     # CLHEP.
-    patch('CLHEP-10.03.03.patch', level=1, when='@10.03.p03')
+    patch('CLHEP-10.03.03.patch', level=1, when='@10.3.3')
+    # These patches can be applied independent of the cxxstd value?
+    patch('cxx17.patch', when='@:10.3.99 cxxstd=17')
+    patch('cxx17_geant4_10_0.patch', level=1, when='@10.4.0 cxxstd=17')
 
     def cmake_args(self):
         spec = self.spec
+        on_or_off = lambda opt: 'ON' if '+' + opt in spec else 'OFF'
 
+        # Core options
         options = [
-            '-DGEANT4_USE_GDML=ON',
+            '-DGEANT4_BUILD_CXXSTD=c++{0}'.format(
+                self.spec.variants['cxxstd'].value),
             '-DGEANT4_USE_SYSTEM_CLHEP=ON',
-            '-DGEANT4_USE_SYSTEM_CLHEP_GRANULAR=ON',
-            '-DGEANT4_USE_G3TOG4=ON',
-            '-DGEANT4_INSTALL_DATA=ON',
-            '-DGEANT4_BUILD_TLS_MODEL=global-dynamic',
             '-DGEANT4_USE_SYSTEM_EXPAT=ON',
             '-DGEANT4_USE_SYSTEM_ZLIB=ON',
-            '-DXERCESC_ROOT_DIR:STRING=%s' %
-            spec['xerces-c'].prefix, ]
+            '-DGEANT4_USE_G3TOG4=ON',
+            '-DGEANT4_USE_GDML=ON',
+            '-DXERCESC_ROOT_DIR={0}'.format(spec['xerces-c'].prefix)
+        ]
 
+        # Multithreading
+        options.append('-DGEANT4_BUILD_MULTITHREADED=' + on_or_off('threads'))
+        if '+threads' in spec:
+            # This should be a variant
+            options.append('-DGEANT4_BUILD_TLS_MODEL=global-dynamic')
+
+        # install the data with geant4
+        datadir = spec['geant4-data'].prefix.share
+        dataver = 'geant4-data-{0}'.format(spec['geant4-data'].version)
```

Comment:
Maybe to be consistent with the setup in geant4-data?
```suggestion
        dataver = '{0}-{1}'.format(spec['geant4-data']..name, spec['geant4-data'].version.dotted)
```

Also I wonder, if geant4-data could have a property/value that could be used here?